src/devices/Common/Iot/Device/Common/AngleExtensions.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+using UnitsNet;
+
+namespace Iot.Device.Common
+{
+    /// <summary>
+    /// Provides extension methods for <see cref="UnitsNet.Angle"/>
+    /// </summary>
+    public static class AngleExtensions
+    {
+        /// <summary>
+        /// Normalizes the angle so it is between 0° and 360° or between -180° and +180° respectively.
+        /// </summary>
+        /// <param name="self">Instance to normalize</param>
+        /// <param name="toFullCircle">Set to true to normalize to 0-360°, otherwise normalizes to +/-180°</param>
+        public static Angle Normalize(this Angle self, bool toFullCircle)
+        {
+            double r = self.Radians;
+            if (toFullCircle)
+            {
+                if (r > Math.PI * 2)
+                {
+                    r = r % (Math.PI * 2);
+                }
+
+                if (r < 0)
+                {
+                    r = -(Math.Abs(r) % (Math.PI * 2));
+                    if (r < 0)
+                    {
+                        r += Math.PI * 2;
+                    }
+                }
+            }
+            else
+            {
+                if (r > Math.PI)
+                {
+                    r = r % (Math.PI * 2);
+                    if (r > Math.PI)
+                    {
+                        // Still above 180?
+                        r -= Math.PI * 2;
+                    }
+                }
+
+                if (r < -Math.PI)
+                {
+                    r = -(Math.Abs(r) % (Math.PI * 2));
+                    if (r < -Math.PI)
+                    {
+                        r += Math.PI * 2;
+                    }
+                }
+            }
+
+            // Return in same unit as original input
+            return Angle.FromRadians(r).ToUnit(self.Unit);
+        }
+
+        /// <summary>
+        /// Calculate the difference between two angles. Useful to compute the angle error between a desired and an actual track.
```

Comment:
you should mention which direction you're calculating the difference

Reply:
Improved documentation